lib/version_utils.pm

```diff
@@ -394,6 +395,11 @@ sub is_using_system_role_first_flow {
     return is_leap('=15.0') || is_using_system_role;
 }
 
+# If there is only one role, there is no selection offered
+sub requires_role_selection {
```

Comment:
That is exactly what `is_using_system_role` is doing. I thought you were fine to add new condition to `is_using_system_role` + change to piece of code where we fallback to desktop selection for openSUSE and exclude krypton from there.

Reply:
That's what I did initially and it's wrong - for openSUSE `is_using_system_role` has a different meaning. If it returns false, the old desktop selection is used instead.

Reply:
It's not the case if you don't trigger "load_system_role_tests" in first place, secondly, it's not that Krypton doesn't have system roles, it just has one, so we should call `load_system_role_tests`. That `elsif` for opensuse is the problem. `is_using_system_role` is opposite of `requires_role_selection` so it's duplication.

Reply:
> It's not the case if you don't trigger "load_system_role_tests" in first place

It is.

> is_using_system_role is opposite of requires_role_selection so it's duplication.

Wrong: 42.3 has neither.

Reply:
As I said, that's how I implemented it in the old PR and it just did not work: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/6278

A new function is necessary.

Reply:
Once again `is_using_system_role` serves same purpose as  `requires_role_selection`, but for SLE, so I would keep change in #6278, it's 100% correct. To solve your issue we need to fix wrong assumption we have for openSUSE, which is that if there are no system roles, we have desktop selection, so we either need to fix condition here:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/main_common.pm#L943 or in `load_system_role_tests`, With current proposal we will branch same thing and answer to the same question twice.

Reply:
`    is_using_system_role is opposite of requires_role_selection so it's duplication.

Wrong: 42.3 has neither.
`
+1, my bad, it's exactly same, not opposite.

Reply:
> so we either need to fix condition here:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/main_common.pm#L943

That's what this patch does though.

There are three possible cases which have to be accounted for:

* Desktop selection
* Role selection
* Neither

It's mathematically impossible to do that without introducing another function.

Reply:
I understand why "is_using_system_role" is not covering this behaviour but I understand it comes from either the version of the yast installer used or settings in control.xml, correct? But why would Krypton be different than all other products?

Reply:
Krypton uses a customized version of TW's control.xml.

Reply:
and you want to preserve that difference?

Reply:
Yes.